gherkin-utils/java/gherkin-utils/src/main/java/gherkin/utils/formatter/GherkinPrettyFormatter.java

```diff
@@ -0,0 +1,101 @@
+package gherkin.utils.formatter;
+
+import gherkin.utils.formatter.model.BackgroundStepContainer;
+import gherkin.utils.formatter.model.IStepContainer;
+import gherkin.utils.formatter.model.ScenarioStepContainer;
+import io.cucumber.messages.Messages.GherkinDocument;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class GherkinPrettyFormatter {
+    public static String pretty(GherkinDocument gherkinDocument) {
+
+        GherkinDocument.Feature feature = gherkinDocument.getFeature();
+        String featureData = prettyTags(feature.getTagsList(), "");
+
+        featureData += feature.getKeyword() + ": " + feature.getName() + "\n";
+        if (!feature.getDescription().isEmpty()) {
+            featureData += feature.getDescription() + "\n";
+        }
+
+        for (GherkinDocument.Feature.FeatureChild featureChild : feature.getChildrenList()) {
+            if (featureChild.hasBackground()) {
+                featureData += prettyStepContainer(new BackgroundStepContainer(featureChild.getBackground()), "  ");
+            } else if (featureChild.hasScenario()) {
+                featureData += prettyStepContainer(new ScenarioStepContainer(featureChild.getScenario()), "  ");
+            } else if (featureChild.hasRule()) {
+                GherkinDocument.Feature.FeatureChild.Rule featureChildRule = featureChild.getRule();
+                featureData += "\n  " + featureChildRule.getKeyword() + ": " + featureChildRule.getName() + "\n";
+                if (!featureChildRule.getDescription().isEmpty()) {
+                    featureData += featureChildRule.getDescription() + "\n";
+                }
+
+                for (GherkinDocument.Feature.FeatureChild.RuleChild ruleChild : featureChildRule.getChildrenList()) {
+                    if (ruleChild.hasBackground()) {
+                        featureData += prettyStepContainer(new BackgroundStepContainer(ruleChild.getBackground()), "    ");
+                    }
+
+                    if (ruleChild.hasScenario()) {
+                        featureData += prettyStepContainer(new ScenarioStepContainer(ruleChild.getScenario()), "    ");
+                    }
+                }
+            }
+        }
+
+        return featureData;
+    }
+
+    public static String prettyTags(List<GherkinDocument.Feature.Tag> tags, String indent) {
```

Comment:
All methods that are not part of the public API should be `private` or package private.

Reply:
Sr, this is my mistake. I have already fixed.